src/Core/src/Platform/Gtk/MauiDatePicker.cs

```diff
@@ -1,35 +1,443 @@
+			PackEnd(_button, false, false, 0);
+		}
+
+		public Gtk.Entry Entry
+		{
+			get
+			{
+				return _entry;
+			}
+		}
+
+		public Gtk.Button PopupButton
+		{
+			get
+			{
+				return _button;
+			}
+		}
+
+		public Color Color
+		{
+			get { return _color; }
+			set
+			{
+				_color = value;
+				Entry.UpdateTextColor(_color);
+			}
+		}
+
+		public void SetBackgroundColor(Gdk.Color color)
+		{
+			var redColor = new Gdk.Color(255, 0, 0);
+			var blueColor = new Gdk.Color(0, 0, 255);
+			ModifyBg(Gtk.StateType.Normal, redColor);
+			Entry.OverrideBackgroundColor(Gtk.StateFlags.Normal, Colors.Blue.ToGdkRgba());
+		}
+	}
+
+	public class MauiDatePicker : EventBox
+	{
+		readonly CustomComboBox _comboBox;
+		DateTime _currentDate;
+		string? _dateFormat;
+		DatePickerWindow? _pickerWindow;
+
+		public event EventHandler DateChanged = new EventHandler((sender, args) => { });
+		public event EventHandler GotFocus = new EventHandler((sender, args) => { });
+		public event EventHandler LostFocus = new EventHandler((sender, args) => { });
+
+		public MauiDatePicker()
+		{
+			_comboBox = new CustomComboBox();
+			Add(_comboBox);
+
+			if ((Child != null))
+				Child.ShowAll();
+			Show();
 
-#pragma warning disable 169
-		// to simulate
-		// Tapping either of the DatePicker displays invokes the platform date picker
-		Gtk.Calendar? _calendar;
-#pragma warning restore 169
+			Date = DateTime.Now;
+			Format = string.Empty;
+			MinDate = new DateTime(1900, 1, 1);
+			MaxDate = new DateTime(2199, 1, 1);
 
-		DateTime _time;
+			_comboBox.Entry.CanDefault = false;
+			_comboBox.Entry.CanFocus = false;
+			_comboBox.Entry.IsEditable = false;
+			_comboBox.Entry.SetStateFlags(StateFlags.Normal, true);
+			_comboBox.Entry.CanFocus = false;
+			_comboBox.Entry.IsEditable = false;
+			_comboBox.Entry.FocusGrabbed += new EventHandler(OnEntryFocused);
+			_comboBox.PopupButton.Clicked += new EventHandler(OnBtnShowCalendarClicked);
+		}
 
 		public DateTime Date
 		{
-			get => _time;
+			get
+			{
+				return _currentDate;
+			}
 			set
 			{
-				_time = value;
-				Text = _time.ToString();
+				_currentDate = value;
+				UpdateEntryText();
 			}
 		}
 
-		public string Format { get; set; }
+		public DateTime MinDate { get; set; }
+
+		public DateTime MaxDate { get; set; }
 
+		public string? Format
+		{
+			get
+			{
+				return _dateFormat;
+			}
+			set
+			{
+				_dateFormat = value;
+				UpdateEntryText();
+			}
+		}
+
+		public void SetBackgroundColor(Gdk.Color color)
+		{
+			_comboBox.SetBackgroundColor(color);
+		}
+
+		void ShowPickerWindow()
+		{
+			int x = 0;
+			int y = 0;
+
+			Window.GetOrigin(out x, out y);
+			y += Allocation.Height;
+
+			var picker = new DatePickerWindow();
+			picker.Move(x, y);
+			picker.SelectedDate = Date;
+			picker.MinimumDate = MinDate;
+			picker.MaximumDate = MaxDate;
+			picker.OnDateTimeChanged += OnPopupDateChanged;
+			picker.Destroyed += OnPickerClosed;
+
+			_pickerWindow = picker;
```

Comment:
@parhamsaremi what if ShowPickerWindow() is called twice, you're not discarding the previous picker? is this how it's done in XamarinForms' DatePicker?

Reply:
They are always creating a new picker and showing that too:
https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Platform.GTK/Controls/DatePicker.cs#L347

we just saved the picker window to use it later :thinking:. I don't exactly remember the reason if you want I can test removing that too.